### PR TITLE
Add io.github.heathcliff26.go-minesweeper

### PR DESCRIPTION
**Pull request by heathcliff26**
### Please confirm your submission meets all the criteria

This is a minesweeper implementation using fyne.io. I mainly created it because i wanted to built a GUI app.

- [x] Please describe the application briefly. 
- [x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link: https://github.com/heathcliff26/go-minesweeper**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.heathcliff26.go-minesweeper.desktop
Comment (hfiguiere) [inline io.github.heathcliff26.go-minesweeper.desktop:1]: this should be managed upstream

File: io.github.heathcliff26.go-minesweeper.metainfo.xml
Comment (hfiguiere) [inline io.github.heathcliff26.go-minesweeper.metainfo.xml:1]: this should be managed upstream

File: io.github.heathcliff26.go-minesweeper.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          version-query: .tag_name
+          timestamp-query: .published_at
+          url-query: '"https://raw.githubusercontent.com/heathcliff26/go-minesweeper/refs/tags/" + $tag + "/img/mine.png"'
+      - type: file
+        path: io.github.heathcliff26.go-minesweeper.desktop
+      - type: file
+        path: io.github.heathcliff26.go-minesweeper.metainfo.xml
+
+    build-commands:
+      - tar -xzf go-minesweeper.tar.gz
```
Comment (hfiguiere) [inline io.github.heathcliff26.go-minesweeper.yaml:55]: source should be of type: `archive` instead

File: io.github.heathcliff26.go-minesweeper.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  <releases>
+    <release version="v0.5.5" date="2024-11-01" type="stable">
+      <description>
+        <p>Dependency bumps</p>
+      </description>
+      <url type="details">https://github.com/heathcliff26/go-minesweeper/releases/tag/v0.5.5</url>
+    </release>
+  </releases>
+
```
[the remark this answers is not in the document]
Comment (heathcliff26) [reply io.github.heathcliff26.go-minesweeper.metainfo.xml:37]: Should the screenshots be managed here or upstream?

File: io.github.heathcliff26.go-minesweeper.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        x-checker-data:
+          is-main-source: true
+          type: json
+          url: https://api.github.com/repos/heathcliff26/go-minesweeper/releases/latest
+          version-query: .tag_name
+          timestamp-query: .published_at
+          url-query: .assets | map(.browser_download_url | select(endswith("_linux-arm64.tar.gz"))) | first
+      - type: file
+        dest-filename: go-minesweeper.png
+        url: https://raw.githubusercontent.com/heathcliff26/go-minesweeper/refs/tags/v0.5.5/img/mine.png
```
Comment (hfiguiere) [inline io.github.heathcliff26.go-minesweeper.yaml:40]: instead this should be in the tarball

File: io.github.heathcliff26.go-minesweeper.metainfo.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply io.github.heathcliff26.go-minesweeper.metainfo.xml:37]: upstream, always

File: io.github.heathcliff26.go-minesweeper.desktop
Comment (heathcliff26) [reply io.github.heathcliff26.go-minesweeper.desktop:1]: Done

File: io.github.heathcliff26.go-minesweeper.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          is-main-source: true
+          type: json
+          url: https://api.github.com/repos/heathcliff26/go-minesweeper/releases/latest
+          version-query: .tag_name
+          timestamp-query: .published_at
+          url-query: '"https://raw.githubusercontent.com/heathcliff26/go-minesweeper/refs/tags/" + $tag + "/io.github.heathcliff26.go-minesweeper.metainfo.xml"'
+
+    build-commands:
+      - install -D go-minesweeper $FLATPAK_DEST/bin/go-minesweeper
+      - desktop-file-edit --set-key Icon --set-value $FLATPAK_ID $FLATPAK_ID.desktop
```
Comment (hfiguiere) [inline io.github.heathcliff26.go-minesweeper.yaml:50]:
```suggestion
      - desktop-file-edit --set-icon=$FLATPAK_ID $FLATPAK_ID.desktop
```

File: io.github.heathcliff26.go-minesweeper.metainfo.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (heathcliff26) [reply io.github.heathcliff26.go-minesweeper.metainfo.xml]: I have created screenshots upstream and the metainfo passes the linter. However with the screenshots the output of the linter changed to `? Validation was successful: pedantic: 3`.
Is that ok, and what does it mean? It seems the linter has no verbose option and i did not find anything to it scanning the docs.

File: io.github.heathcliff26.go-minesweeper.yaml (hunk 1 above)
Comment (heathcliff26) [reply io.github.heathcliff26.go-minesweeper.yaml]: done

File: io.github.heathcliff26.go-minesweeper.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        x-checker-data:
+          is-main-source: true
+          type: json
+          url: https://api.github.com/repos/heathcliff26/go-minesweeper/releases/latest
+          version-query: .tag_name
+          timestamp-query: .published_at
+          url-query: '"https://raw.githubusercontent.com/heathcliff26/go-minesweeper/refs/tags/" + $tag + "/io.github.heathcliff26.go-minesweeper.metainfo.xml"'
+
+    build-commands:
+      - install -D go-minesweeper $FLATPAK_DEST/bin/go-minesweeper
```
Comment (hfiguiere) [inline io.github.heathcliff26.go-minesweeper.yaml:49]:
```suggestion
      - install -Dm755 go-minesweeper $FLATPAK_DEST/bin/go-minesweeper
```

File: io.github.heathcliff26.go-minesweeper.yaml (hunk 3 above)
Comment (heathcliff26) [reply io.github.heathcliff26.go-minesweeper.yaml]: done

**Comment by hfiguiere** [comment]
bot, build io.github.heathcliff26.go-minesweeper

File: io.github.heathcliff26.go-minesweeper.yaml (hunk 4 above)
Comment (heathcliff26) [reply io.github.heathcliff26.go-minesweeper.yaml]: done

**Comment by bbhtt** [comment]
This needs a better icon, it's completely black which makes it almost invisible when the DE has dark mode turned on.

**Comment by heathcliff26** [comment]
Thank you @hfiguiere for the review and help. I think this was one of the quickest reviews i have ever experienced.

**Pull request closed without merging**